### PR TITLE
[#434] feat: added create topic endpoint

### PR DESCRIPTION
**Pull request by ilkerkocatepe**
This PR resolves issue https://github.com/strimzi/strimzi-kafka-bridge/issues/434

First of all, I am very sorry for the delay. From now on, I will make sure to take a task when I can complete it quickly.

### CODE REVIEW

**Comment by ilkerkocatepe** [comment]
Hello @ppatierno, I have an issue with build docs, do you have an idea regarding fix it?

**Comment by ppatierno** [comment]
> Hello @ppatierno, I have an issue with build docs, do you have an idea regarding fix it?

@ilkerkocatepe we have issues with swagger2markup. Some of its dependencies are disappeared from JCenter (and I also read that JCenter was dismissed by JFrog and all calls are redirected to Maven Central). I am investigating about a replacement for swagger2markup (maype using openapi-generator) but it's not there yet, so the build problem with doc is expected right now. I am sorry.

**Comment by ilkerkocatepe** [comment]
> > Hello @ppatierno, I have an issue with build docs, do you have an idea regarding fix it?
> 
> @ilkerkocatepe we have issues with swagger2markup. Some of its dependencies are disappeared from JCenter (and I also read that JCenter was dismissed by JFrog and all calls are redirected to Maven Central). I am investigating about a replacement for swagger2markup (maype using openapi-generator) but it's not there yet, so the build problem with doc is expected right now. I am sorry.

thank you @ppatierno could you review the pr?

File: src/main/java/io/strimzi/kafka/bridge/KafkaBridgeAdmin.java [hunk 1]
```diff
@@ -84,6 +86,29 @@ public CompletionStage<Set<String>> listTopics() {
...
+     * Creates a topic with given name
+     *
+     * @param topicName topic name to create
+     * @return a CompletionStage Void
+     */
+    public CompletionStage<Void> createTopic(String topicName) {
+        LOGGER.trace("Create topic thread {}", Thread.currentThread());
+        LOGGER.info("Create topic {}", topicName);
+        CompletableFuture<Void> promise = new CompletableFuture<>();
+        this.adminClient.createTopics(Collections.singletonList(new NewTopic(topicName, 2, (short) 1)))
```
Comment (ppatierno) [inline src/main/java/io/strimzi/kafka/bridge/KafkaBridgeAdmin.java:99]: why are partitions and replication factor hard coded? They should be parameters passed through the API imho.

File: src/main/resources/openapi.json [hunk 2]
```diff
@@ -753,6 +753,56 @@
                 }
             ]
         },
+        "/create-topic/{topicname}": {
```
Comment (ppatierno) [inline src/main/resources/openapi.json:756]: I don't think this API design is going to work. What are you going to have if we want the topic deletion, a new `/delete-topic` endpoint? It's not how REST API design works and scales.
I think we would need a new more general `/admin` endpoint? So a corresponding `/admin/topics` on which we can use the usual POST to create, DELETE to delete and so on? I think you should think more broader than just adding the topic creation, in terms of the API evolution for additional operations.

**Review by ppatierno — CHANGES_REQUESTED** [review]

**Comment by ppatierno** [comment]
@ilkerkocatepe other than taking a look at the comments I left, you could rebase against the main. We have fixed the build documentation related issue (now using a new maven plugin).

File: src/main/java/io/strimzi/kafka/bridge/KafkaBridgeAdmin.java (hunk 1 above)
Comment (ilkerkocatepe) [reply src/main/java/io/strimzi/kafka/bridge/KafkaBridgeAdmin.java:99]: Sorry I missed this, will be fixed 👍

File: src/main/resources/openapi.json (hunk 2 above)
Comment (ilkerkocatepe) [reply src/main/resources/openapi.json:756]: You're right, I thought about it but couldn't decide, I will do as you said.

**Comment by ppatierno** [comment]
@ilkerkocatepe I see you rebased but building the doc failed with this ...

```shell
ERROR: Uncommitted changes in documentation:
M	documentation/book/api/.openapi-generator/openapi.json-generate-apidoc.sha256
M	documentation/book/api/index.adoc
Run the following to add up-to-date resources:
  make docu_api \
    && git add documentation/book/ \
    && git commit -s -m 'Update generated documentation'
```

Of course this is because you are changing the OpenAPI spec so the new openapi-generator is generatic a new index.adoc (with a different SHA) which you didn't push.
There is no need to push them now even because, as agreed, you are going to change the OpenAPI spec again so remember just to push the new above files when the new implementation is done.

**Comment by ilkerkocatepe** [comment]
@ppatierno 
I finished the implementation that we have agreed. Could you check again if it's okay?
I pushed new doc files but it's conflicted

**Comment by ppatierno** [comment]
@ilkerkocatepe did you rebase against main before pushing your changes?

**Comment by ppatierno** [comment]
@ilkerkocatepe I see the PR in a little mess ... I don't think you changed 58 files :-D It contains all the other commits coming from my fix on the bridge doc and they should not be there. You should fix this first (I think the conflict will go away) and then I can properly review. Thanks!

**Comment by ilkerkocatepe** [comment]
> @ilkerkocatepe did you rebase against main before pushing your changes?

of course but something went wrong :(
I'm gonna open a clear pr :)

**Pull request closed without merging**